File: java/src/main/java/com/languageweaver/sdk/samples/languagedetection/file/EdgeFileLanguageDetectionService.java
package com.languageweaver.sdk.samples.languagedetection.file;

import com.languageweaver.sdk.common.constants.Format;
import com.languageweaver.sdk.common.edge.EdgeLanguageWeaverClient;
import com.languageweaver.sdk.languagedetection.edge.request.EdgeFileLanguageDetectionRequest;
import com.languageweaver.sdk.languagedetection.edge.result.EdgeLanguageDetectionResult;

import java.nio.file.Paths;

public class EdgeFileLanguageDetectionService {
    public static void main(String[] args) throws Exception {
        try (EdgeLanguageWeaverClient lwClient = new EdgeLanguageWeaverClient().build()) {
            EdgeFileLanguageDetectionRequest edgeFileLanguageDetectionRequest = new EdgeFileLanguageDetectionRequest()
                    .setInputFile(Paths.get("java", "src", "main", "resources", "input", "input1.txt").toFile().getAbsolutePath())
                    .setInputFormat(Format.PLAIN)
                    .setEncoding("UTF-8");

            final EdgeLanguageDetectionResult edgeLanguageDetectionResult = lwClient.detectLanguage(edgeFileLanguageDetectionRequest);

            if (edgeLanguageDetectionResult.getLanguages() != null) {
                edgeLanguageDetectionResult.getLanguages()
                        .forEach(language -> System.out.println(language.getName()));
            }
        }
    }
}

File: javascript/browser/src/samples/language-detection/file/EdgeFileLanguageDetectionService.js
import {
    ClientConfiguration,
    CredentialsConfiguration,
    EdgeFileLanguageDetectionRequest,
    EdgeLanguageWeaverClient,
    Format
} from "@language-weaver/lw-sdk-js";
import {file1} from "../../../samples-utils";

export const detectEdgeFileLanguage = async (clientId) => {
    try {
        const clientConfiguration = new ClientConfiguration();
        clientConfiguration.credentialsConfiguration = new CredentialsConfiguration(clientId);
        const edgeLanguageWeaverClient = await new EdgeLanguageWeaverClient()
            .withConfigurations(clientConfiguration)
            .build();
        const edgeFileLanguageDetectionRequest = new EdgeFileLanguageDetectionRequest();
        edgeFileLanguageDetectionRequest.inputFile = file1;
        edgeFileLanguageDetectionRequest.inputFormat = Format.PLAIN;

        const edgeFileLanguageDetectionResult = await edgeLanguageWeaverClient.detectEdgeFileLanguage(edgeFileLanguageDetectionRequest);
        console.log(edgeFileLanguageDetectionResult);
    } catch (e) {
        console.log(e);
    }
}